src/state/opamSwitchState.ml

```diff
@@ -392,11 +398,12 @@ let load lock_kind gt rt switch =
     else switch_config
   in
   let switch_config, switch_invariant =
-    if OpamVersion.compare
-        switch_config.OpamFile.Switch_config.opam_version
-        (OpamVersion.of_string "2.1")
-       >= 0
-    || switch_config.OpamFile.Switch_config.invariant <> OpamFormula.Empty
+  (* tweak this check, but with what ? *)
+    if (match OpamFile.Config.opam_root_version gt.config with
+        | None -> false
+        | Some c ->
+          OpamVersion.compare c (OpamVersion.of_string "2.1.0~rc") >= 0)
+    && switch_config.OpamFile.Switch_config.invariant <> OpamFormula.Empty
```

Comment:
This should still be `||`, right - we don't need to recompute the switch invariant if there is one there?

Reply:
fixed with invariant being an option, if undefined it is clearly at `None`